test/controllers/api/v2/hosts_controller_test.rb

```diff
@@ -108,6 +108,17 @@ def expect_attribute_modifier(modifier_class, args)
     assert_equal Host.all.pluck(:id, :name), hosts['results'].map(&:values)
   end
 
+  test "subtotal should be the same as the search count with thin" do
+    FactoryBot.create_list(:host, 2)
+    Host.last.update_attribute(:name, 'test')
+
+    get :index, params: { thin: true, per_page: 1, search: 'host' }
+    assert_response :success
+    assert_not_nil assigns(:hosts)
+    hosts = ActiveSupport::JSON.decode(@response.body)
+    assert_equal hosts['subtotal'], Host.search_for('host').size
```

Comment:
there might only be 1 host matching the search so this test will pass even before the test (i think you are relaying on the hosts in fixtures to have a longer result)

Reply:
There're two matched hosts, including the host from the setup:
```ruby
# /test/controllers/api/v2/hosts_controller_test.rb
 def setup
    as_admin do
      @host = FactoryBot.create(:host)
```